CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+    src/ripple/crypto/csprng.h
+  DESTINATION include/ripple/crypto)
+install (
+  FILES
+    src/ripple/crypto/impl/ec_key.h
+    src/ripple/crypto/impl/openssl.h
+  DESTINATION include/ripple/crypto/impl)
+install (
+  FILES
+    src/ripple/json/JsonPropertyStream.h
+    src/ripple/json/Object.h
+    src/ripple/json/Output.h
+    src/ripple/json/Writer.h
+    src/ripple/json/json_forwards.h
+    src/ripple/json/json_reader.h
+    src/ripple/json/json_value.h
+    src/ripple/json/json_writer.h
+    src/ripple/json/to_string.h
+  DESTINATION include/ripple/json)
+install (
+  FILES
+    src/ripple/json/impl/json_assert.h
+  DESTINATION include/ripple/json/impl)
+install (
+  FILES
+    src/ripple/protocol/AccountID.h
+    src/ripple/protocol/AmountConversions.h
+    src/ripple/protocol/Book.h
+    src/ripple/protocol/BuildInfo.h
+    src/ripple/protocol/ErrorCodes.h
+    src/ripple/protocol/Feature.h
+    src/ripple/protocol/HashPrefix.h
+    src/ripple/protocol/IOUAmount.h
+    src/ripple/protocol/Indexes.h
+    src/ripple/protocol/InnerObjectFormats.h
+    src/ripple/protocol/Issue.h
+    src/ripple/protocol/JsonFields.h
+    src/ripple/protocol/Keylet.h
+    src/ripple/protocol/KnownFormats.h
+    src/ripple/protocol/LedgerFormats.h
+    src/ripple/protocol/Protocol.h
+    src/ripple/protocol/PublicKey.h
+    src/ripple/protocol/Quality.h
+    src/ripple/protocol/Rate.h
+    src/ripple/protocol/SField.h
+    src/ripple/protocol/SOTemplate.h
+    src/ripple/protocol/STAccount.h
+    src/ripple/protocol/STAmount.h
+    src/ripple/protocol/STArray.h
+    src/ripple/protocol/STBase.h
+    src/ripple/protocol/STBitString.h
+    src/ripple/protocol/STBlob.h
+    src/ripple/protocol/STExchange.h
+    src/ripple/protocol/STInteger.h
+    src/ripple/protocol/STLedgerEntry.h
+    src/ripple/protocol/STObject.h
+    src/ripple/protocol/STParsedJSON.h
+    src/ripple/protocol/STPathSet.h
+    src/ripple/protocol/STTx.h
+    src/ripple/protocol/STValidation.h
+    src/ripple/protocol/STVector256.h
+    src/ripple/protocol/SecretKey.h
+    src/ripple/protocol/Seed.h
+    src/ripple/protocol/Serializer.h
+    src/ripple/protocol/Sign.h
+    src/ripple/protocol/SystemParameters.h
+    src/ripple/protocol/TER.h
+    src/ripple/protocol/TxFlags.h
+    src/ripple/protocol/TxFormats.h
+    src/ripple/protocol/UintTypes.h
+    src/ripple/protocol/XRPAmount.h
+    src/ripple/protocol/digest.h
+    src/ripple/protocol/tokens.h
+  DESTINATION include/ripple/protocol)
+install (
+  FILES
+    src/ripple/protocol/impl/STVar.h
+    src/ripple/protocol/impl/secp256k1.h
+  DESTINATION include/ripple/protocol/impl)
+
+#[===================================================================[
+   rippled executable
+#]===================================================================]
+
+#[=========================================================[
+   this one header is added as source just to keep older
+   versions of cmake happy. cmake 3.10+ allows
```

Comment:
There are several nice features that came in 3.10 and 3.11 family. Any particular reason why 3.7 was chosen as the min version to support?